### PR TITLE
Make avatar depending

### PR DESCRIPTION
**Pull request by DennisMaas**
It like to show different avatar icons depending on the type of photography. this is why i created IconSwitcher.js, but just putting `<IconSwitcher/> ` into line 26 in PlaceCard.js did not work.
I can not figure out how to put it into it´s own js-component, so this all happens inline and on the top level. I could not get it to work inside the avatar (see line 25 ff in the old code) like it did when i just had to display one. Please leave a comment if you know how to do this more elegantly.

### CODE REVIEW

File: frontend/src/components/cards/IconSwitcher.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+export default function IconSwitcher() {
+  const { id } = useParams();
+  const { places } = useContext(PlacesContext);
+  const placeData = places.find((place) => place.id === id);
+
+  switch (placeData.type) {
+    case 'architecture':
+      return <ArchitectureIcon />;
+      break;
```
Comment (jamarob) [inline frontend/src/components/cards/IconSwitcher.js:18]: When you return from a switch, you don#t need the break.

File: frontend/src/components/cards/IconSwitcher.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+import ArchitectureIcon from './avatarIcons/ArchitectureIcon';
+import HarbourIcon from './avatarIcons/HarbourIcon';
+import NightIcon from './avatarIcons/NightIcon';
+import ImageIcon from './avatarIcons/ImageIcon';
+import PlacesContext from '../../contexts/PlacesContext';
+
+export default function IconSwitcher() {
+  const { id } = useParams();
+  const { places } = useContext(PlacesContext);
+  const placeData = places.find((place) => place.id === id);
```
Comment (jamarob) [inline frontend/src/components/cards/IconSwitcher.js:13]: Overall, this is way too complicated. Why not simply a function that get's the type of your place and returns the icon?

Comment (DennisMaas) [reply frontend/src/components/cards/IconSwitcher.js:13]: my switch was intended to be such a simple function. :-(

**Comment by jamarob** [comment]
Your frontend build fails because your App.test.js fails, fix the query string to make it pass.

**Comment by DennisMaas** [comment]
> Your frontend build fails because your App.test.js fails, fix the query string to make it pass.

```
import { render, screen } from '@testing-library/react';
import App from './App';

test('renders learn react link', () => {
  render(<App />);
  const linkElement = screen.getByText(/learn react/i);
  expect(linkElement).toBeInTheDocument();
});
```

Wich part is the query? I did not touch it in any way, i think.

**Comment by jamarob** [comment]
> Wich part is the query? I did not touch it in any way, i think.

https://testing-library.com/docs/react-testing-library/cheatsheet#queries

File: frontend/src/components/cards/IconSwitcher.js (hunk 2 above)
Comment (DennisMaas) [reply frontend/src/components/cards/IconSwitcher.js:13]: i wrote my switch in iconswitcher.js without breaks now, but this leaves me with a TypeError: Cannot read property 'type' of undefined and i am stuck again

**Pull request closed without merging**